### PR TITLE
Merge changes from dd-trace-java 0.58.0, 0.59.0, 0.60.0 (Part 1)

### PR DESCRIPTION
**Pull request by trask**
https://github.com/DataDog/dd-trace-java/releases/tag/v0.58.0
https://github.com/DataDog/dd-trace-java/releases/tag/v0.59.0
https://github.com/DataDog/dd-trace-java/releases/tag/v0.60.0

Thanks @richardstartin @tylerbenson @randomanderson!

All of the original PRs are linked in the associated commit messages.

---------------
Breaking these commits out from the (too) big PR #1089 (which is now renamed Part 2)

There are still a lot of commits in this PR, but they seemed to be the ones with no outstanding review comments.

I will try to do smaller merge commits going forwards.

### CODE REVIEW

File: instrumentation/scala-concurrent/scala-concurrent.gradle [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+apply from: "$rootDir/gradle/test-with-scala.gradle"
+apply plugin: 'org.unbroken-dome.test-sets'
+
+muzzle {
+  pass {
+    group = 'org.scala-lang'
+    module = "scala-library"
+    versions = "[2.8.0,2.12.0)"
+    assertInverse = true
+  }
```
Comment (trask) [inline instrumentation/scala-concurrent/scala-concurrent.gradle:11]: cc: @tylerbenson, it looks like muzzle in dd-trace-java is only ever checking `2.11.12` b/c of

```
resolutionStrategy {
  force deps.scala
}
```

in `test-with-scala.gradle`

File: instrumentation/akka-context-propagation-2.5/src/main/java/io/opentelemetry/instrumentation/auto/akkaconcurrent/AkkaForkJoinPoolInstrumentation.java [hunk 2]
```diff
@@ -28,27 +28,33 @@
...
   }
 
   @Override
   protected boolean defaultEnabled() {
     return false;
   }
 
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    // This might need to be an extendsClass matcher...
```
Comment (anuraaga) [inline instrumentation/akka-context-propagation-2.5/src/main/java/io/opentelemetry/instrumentation/auto/akkaconcurrent/AkkaForkJoinPoolInstrumentation.java:54]: Can we confirm this caveat? Should we file an issue?

**Review by anuraaga — APPROVED** [review]

File: instrumentation/akka-context-propagation-2.5/src/main/java/io/opentelemetry/instrumentation/auto/akkaconcurrent/AkkaForkJoinPoolInstrumentation.java (hunk 2 above)
Comment (trask) [reply instrumentation/akka-context-propagation-2.5/src/main/java/io/opentelemetry/instrumentation/auto/akkaconcurrent/AkkaForkJoinPoolInstrumentation.java:54]: Just checking you saw my response https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1089#discussion_r476619285

File: instrumentation/finatra-2.9/src/latestDepTest/groovy/FinatraServerLatestTest.groovy [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  @Override
+  void handlerSpan(TraceAssert trace, int index, Object parent, String method = "GET", ServerEndpoint endpoint = SUCCESS) {
+    trace.span(index) {
+      operationName "FinatraController"
+      spanKind INTERNAL
+      childOf(parent as SpanData)
+      // Finatra doesn't propagate the stack trace or exception to the instrumentation
+      // so the normal errorAttributes() method can't be used
```
Comment (iNikem) [inline instrumentation/finatra-2.9/src/latestDepTest/groovy/FinatraServerLatestTest.groovy:102]: So how exceptions are recorded in Finatra? Or are they not?

File: instrumentation/guava-10.0/src/main/java/io/opentelemetry/instrumentation/auto/guava/ListenableFutureInstrumentation.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  public static class AddListenerAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static State addListenerEnter(
+        @Advice.Argument(value = 0, readOnly = false) Runnable task,
+        @Advice.Argument(1) final Executor executor) {
+      final Context context = Context.current();
+      final Runnable newTask = RunnableWrapper.wrapIfNeeded(task);
+      // It is important to check potentially wrapped task if we can instrument task in this
+      // executor. Some executors do not support wrapped tasks.
```
Comment (iNikem) [inline instrumentation/guava-10.0/src/main/java/io/opentelemetry/instrumentation/auto/guava/ListenableFutureInstrumentation.java:71]: @trask You have removed this check in another PR, haven't you?

File: instrumentation/hystrix-1.4/src/test/groovy/HystrixObservableChainTest.groovy [hunk 5]
```diff
@@ -19,13 +19,17 @@ import static io.opentelemetry.auto.test.utils.TraceUtils.runUnderTrace
...
 import com.netflix.hystrix.HystrixObservableCommand
 import io.opentelemetry.auto.test.AgentTestRunner
+import io.opentelemetry.auto.test.utils.ConfigUtils
 import rx.Observable
 import rx.schedulers.Schedulers
 
 class HystrixObservableChainTest extends AgentTestRunner {
   static {
     // Disable so failure testing below doesn't inadvertently change the behavior.
     System.setProperty("hystrix.command.default.circuitBreaker.enabled", "false")
```
Comment (iNikem) [inline instrumentation/hystrix-1.4/src/test/groovy/HystrixObservableChainTest.groovy:29]: I think both properties should be inside `ConfigUtils.updateConfig`

File: instrumentation/hystrix-1.4/src/test/groovy/HystrixObservableTest.groovy [hunk 6]
```diff
@@ -30,6 +31,9 @@ class HystrixObservableTest extends AgentTestRunner {
   static {
     // Disable so failure testing below doesn't inadvertently change the behavior.
     System.setProperty("hystrix.command.default.circuitBreaker.enabled", "false")
+    ConfigUtils.updateConfig {
```
Comment (iNikem) [inline instrumentation/hystrix-1.4/src/test/groovy/HystrixObservableTest.groovy:34]: Same here

File: instrumentation/hystrix-1.4/src/test/groovy/HystrixTest.groovy [hunk 7]
```diff
@@ -28,6 +29,9 @@ class HystrixTest extends AgentTestRunner {
   static {
     // Disable so failure testing below doesn't inadvertently change the behavior.
     System.setProperty("hystrix.command.default.circuitBreaker.enabled", "false")
+    ConfigUtils.updateConfig {
```
Comment (iNikem) [inline instrumentation/hystrix-1.4/src/test/groovy/HystrixTest.groovy:32]: And here

File: instrumentation/spring/spring-webflux-5.0/spring-webflux-5.0-library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/SpringWebfluxHttpClientTracer.java [hunk 8]
```diff
@@ -76,4 +87,13 @@ protected String getInstrumentationName() {
   public Tracer getTracer() {
     return tracer;
   }
+
+  private static MethodHandle findRawStatusCode() {
```
Comment (iNikem) [inline instrumentation/spring/spring-webflux-5.0/spring-webflux-5.0-library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/SpringWebfluxHttpClientTracer.java:91]: This certainly needs javadoc

File: javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/test/DefaultInstrumenterTest.groovy [hunk 9]
```diff
@@ -93,9 +93,8 @@ class DefaultInstrumenterTest extends AgentSpecification {
 
   def "configure default sys prop as #value"() {
     setup:
-    ConfigUtils.updateConfig {
-      System.setProperty("otel.integrations.enabled", value)
-    }
+    System.setProperty("otel.integrations.enabled", value)
```
Comment (iNikem) [inline javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/test/DefaultInstrumenterTest.groovy:96]: Why this change? Old code seems better to me :)

File: instrumentation/scala-concurrent/scala-concurrent.gradle (hunk 1 above)
Comment (tylerbenson) [reply instrumentation/scala-concurrent/scala-concurrent.gradle:11]: did you find a way to solve it?

Comment (tylerbenson) [reply instrumentation/scala-concurrent/scala-concurrent.gradle:11]: I guess that's not a problem here since you aren't forcing that version.  Thanks for calling it out.

Comment (tylerbenson) [reply instrumentation/scala-concurrent/scala-concurrent.gradle:11]: I took the liberty: https://github.com/DataDog/dd-trace-java/pull/1810

Comment (tylerbenson) [reply instrumentation/scala-concurrent/scala-concurrent.gradle:11]: It's actually failing for me on `latestDepTestCompile` which you seem to have removed here.

Comment (trask) [reply instrumentation/scala-concurrent/scala-concurrent.gradle:11]: oh, good point, i added it back

File: instrumentation/finatra-2.9/src/latestDepTest/groovy/FinatraServerLatestTest.groovy (hunk 3 above)
Comment (trask) [reply instrumentation/finatra-2.9/src/latestDepTest/groovy/FinatraServerLatestTest.groovy:102]: idk, this is not new code in this PR (copied from `FinatraServerTest.groovy`), opened #1131 to track

File: instrumentation/hystrix-1.4/src/test/groovy/HystrixObservableChainTest.groovy (hunk 5 above)
Comment (trask) [reply instrumentation/hystrix-1.4/src/test/groovy/HystrixObservableChainTest.groovy:29]: `hystrix.command.default.circuitBreaker.enabled` is configuration for the library under test, not for the instrumentation under test, so not sure it should be under `updateConfig`

File: instrumentation/guava-10.0/src/main/java/io/opentelemetry/instrumentation/auto/guava/ListenableFutureInstrumentation.java (hunk 4 above)
Comment (trask) [reply instrumentation/guava-10.0/src/main/java/io/opentelemetry/instrumentation/auto/guava/ListenableFutureInstrumentation.java]: yes, thanks, and this comment prompted #1128

**Review by iNikem — APPROVED** [review]
